src/components/DeckMechanisms/cycle.js

```diff
@@ -1,37 +1,40 @@
-import rwc from 'random-weighted-choice'
 import getIncreasedDeckWeight from './getIncreasedDeckWeight'
+import rwcDuplicates from '../../helpers/rwcDuplicates'
+import isCard from '../../helpers/isCard'
 
 export const DEFAULT_CYCLE_OPTIONS = { countAsCycled: true }
 
 /**
  * Mutate the given state following a cycle.
  * @param {Object} state - State being mutated
- * @param {String} id - Id of the cycled card
+ * @param {Object} card - Object containing id and idx of the cycled card
  * @param {Object} options - Cycling options
  * @param {Boolean} [options.countAsCycled = true] - Whether it counts as this turn’s cycle
  * @return {Object} Mutated state
  */
-const cycle = (state, id, options = DEFAULT_CYCLE_OPTIONS) => {
+const cycle = (state, card, options = DEFAULT_CYCLE_OPTIONS) => {
   // Remove the cycled card from the hand.
-  state.hand = state.hand.filter(cardId => cardId !== id)
+  state.hand = state.hand.filter(cardInHand => !isCard(card)(cardInHand))
```

Comment:
```suggestion
  const isNotCard = card => cardInHand => !isCard(card)(cardInHand)
  state.hand = state.hand.filter(isNotCard(card))
```

You could even consider exporting `isNotCard` from the `isCard` helper.